benchmark/primitives/primitives_test.go

```diff
@@ -425,3 +425,26 @@ func BenchmarkRLockUnlock(b *testing.B) {
 		}
 	})
 }
+
+func BenchmarkEvenBit(b *testing.B) {
+	n := int64(1 << 33)
+	b.ResetTimer()
+	for i := int64(0); i <= n; i++ {
+		runBit(i)
+	}
```

Comment:
This isn't how benchmarks are supposed to work in my experience.

1. you are ignoring b.N, so the measurement won't ever converge.
2. you are performing 2^33 operations for each iteration.  Each iteration should only do the operation once, and let b.N determine how many times you do it.
3. `b.ResetTimer()` and `b.StopTimer()` aren't necessary in these situations.